mne_connectivity/tests/test_wsmi.py

```diff
@@ -0,0 +1,1065 @@
+
+    # Should raise error - single channel connectivity is not meaningful
+    with pytest.raises(ValueError, match="At least 2 channels are required"):
+        wsmi(epochs, kernel=3, tau=1)
+
+
+def test_wsmi_meg_data():
+    """Test with MEG data."""
+    sfreq = 100.0
+    n_epochs, n_channels, n_times = 2, 3, 150
+    data = np.random.RandomState(0).randn(n_epochs, n_channels, n_times)
+
+    # Create MEG channels
+    ch_names = ["MEG0111", "MEG0112", "MEG0113"]
+    info = create_info(ch_names, sfreq=sfreq, ch_types="mag")
+    epochs = EpochsArray(data, info, tmin=0.0)
+
+    # Test with MEG data
+    conn = wsmi(epochs, kernel=3, tau=1)
+
+    assert conn.n_nodes == n_channels
+    assert conn.method == "wSMI"
+
+
+def test_wsmi_all_channels_as_bad():
+    """Test wSMI when all channels are marked as bad."""
+    sfreq = 100.0
+    data = np.random.RandomState(0).randn(2, 2, 100)
+
+    # Create channels and mark them all as bad
+    ch_names = ["Ch1", "Ch2"]
+    info = create_info(ch_names, sfreq=sfreq, ch_types="eeg")
+    epochs = EpochsArray(data, info, tmin=0.0)
+    epochs.info["bads"] = ch_names  # Mark all channels as bad
+
+    # Should process all channels (bad channels are not excluded)
+    result = wsmi(epochs, kernel=3, tau=1)
+
+    # Check that result is valid
+    data_matrix = result.get_data()
+    n_connections = len(ch_names) * (len(ch_names) - 1) // 2
+    assert data_matrix.shape == (2, n_connections)
+    assert np.all(np.isfinite(data_matrix))
+
+
+def test_wsmi_eeg_without_montage():
+    """Test wSMI with EEG data without montage."""
+    sfreq = 100.0
+    n_epochs, n_channels, n_times = 2, 3, 150
+    data = np.random.RandomState(0).randn(n_epochs, n_channels, n_times)
+
+    # Create EEG channels without setting montage
+    ch_names = ["EEG001", "EEG002", "EEG003"]
+    info = create_info(ch_names, sfreq=sfreq, ch_types="eeg")
+    epochs = EpochsArray(data, info, tmin=0.0)
+
+    # Should work fine without any special preprocessing
+    conn = wsmi(epochs, kernel=3, tau=1)
+
+    assert conn.method == "wSMI"
+    assert np.all(np.isfinite(conn.get_data()))
+
+
+@pytest.mark.parametrize("kernel", [2, 3, 4])
+@pytest.mark.parametrize("tau", [1, 2])
+def test_wsmi_parameter_combinations(kernel, tau):
+    """Test various parameter combinations."""
+    sfreq = 100.0
+    n_epochs, n_channels, n_times = 2, 3, 200
+    data = np.random.RandomState(0).randn(n_epochs, n_channels, n_times)
+
+    ch_names = ["Fz", "Cz", "Pz"]
+    info = create_info(ch_names, sfreq=sfreq, ch_types="eeg")
+    epochs = EpochsArray(data, info, tmin=0.0)
+    epochs.set_montage(
+        mne.channels.make_standard_montage("standard_1020"), on_missing="ignore"
+    )
+
+    # Test different kernel and tau combinations
+    conn = wsmi(
+        epochs,
+        kernel=kernel,
+        tau=tau,
+    )
+
+    assert conn.method == "wSMI"
+    assert conn.n_nodes == n_channels
+    assert np.all(np.isfinite(conn.get_data()))
```

Comment:
A test that parametrizes over `kernel` and `tau` values should test that the different `kernel` or `tau` values have different effects. There are at least 2 ways to do that:

1. use non-random channel data, compute from first principles what the conn values should be for different kernel/tau values, pass them in as part of the parametrization, and assert that you get what you expect
2. don't parametrize, run the wsmi a few times (with different kernel/tau values) within the test function, and assert that the resulting `conn.get_data()` arrays are actually different.